docs/source/cpp/streaming_execution.rst

```diff
@@ -305,3 +305,451 @@ Datasets may be scanned multiple times; just make multiple scan
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: OrderBySink Example)
+  :end-before: (Doc section: OrderBySink Example)
+  :linenos:
+  :lineno-match:
+
+
+.. _stream_execution_select_k_docs:
+
+``select_k_sink``
+-----------------
+
+``select_k_sink`` option enables selecting k number of elements. 
+:class:`arrow::compute::SelectKOptions` which is a defined by 
+using :struct:`OrderBySinkNode` definition. This option returns a sink node that receives 
+inputs and then compute top_k/bottom_k.
+
+Create SelectK Option::
+
+  arrow::compute::SelectKOptions options = arrow::compute::SelectKOptions::TopKDefault(
+              /*k=*/2, {"i32"});
+
+  ARROW_ASSIGN_OR_RAISE(
+    arrow::compute::ExecNode * k_sink_node,
+    arrow::compute::MakeExecNode("select_k_sink",
+      plan.get(), {source},
+      arrow::compute::SelectKSinkNodeOptions{options, &sink_gen}));
+
+
+SelectK Example;
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: KSelect Example)
+  :end-before: (Doc section: KSelect Example)
+  :linenos:
+  :lineno-match:
+
+.. _stream_execution_scan_docs:
+
+``scan``
+---------
+
+`scan` is an operation used to load and process data, and the behavior of is defined using 
+:class:`arrow::dataset::ScanNodeOptions`. This option contains a set of definitions. 
+The :ref:`dataset<cpp-dataset-reading>` API also use scanner for processing data.
+In contrast to `source` operation, `scan` operation can load the data and apply scanning
+operations like filter and project to the loaded data. 
+
+Creating a Scan `ExecNode`::
+
+  auto options = std::make_shared<arrow::dataset::ScanOptions>();
+  options->use_async = true; 
+  options->projection = Materialize({});  // create empty projection
+
+  // construct the scan node
+  cp::ExecNode* scan;
+  auto scan_node_options = arrow::dataset::ScanNodeOptions{dataset, options};
+
+  ARROW_ASSIGN_OR_RAISE(scan,
+                          cp::MakeExecNode("scan", plan.get(), {}, 
+                            scan_node_options));
+
+Scan example;
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: Scan Example)
+  :end-before: (Doc section: Scan Example)
+  :linenos:
+  :lineno-match:
+
+.. _stream_execution_write_docs:
+
+``write``
+---------
+
+``write`` option enables writing a result to supported file formats (example `parquet`,
+`feather`, `csv`, etc). 
+The write options are provided via the :class:`arrow::dataset::WriteNodeOptions` and 
+defined using :class:`arrow::dataset::FileSystemDatasetWriteOptions`, 
+``std::shared_ptr<arrow::Schema>``, and 
+``std::shared_ptr<arrow::util::AsyncToggle> backpressure_toggle``. Here the 
+:class:`arrow::dataset::FileSystemDatasetWriteOptions` contains the meta-data required 
+to write the data. 
```

Comment:
```suggestion
The write options are provided via the :class:`arrow::dataset::WriteNodeOptions` and 
defined using :class:`arrow::dataset::FileSystemDatasetWriteOptions`. Here the 
:class:`arrow::dataset::FileSystemDatasetWriteOptions` contains the meta-data required 
to write the data. 
```

Let's hide backpressure for now (that setting is optional I'm pretty sure).  It's actually a bug (my fault) that we require `schema`.  I think we can remove that in the future so let's skip mentioning it for the moment.

Reply:
Got it, updated the doc.